tests/UriTest.php

```diff
@@ -569,14 +569,12 @@ public function testAllowsForRelativeUri()
         $this->assertSame('foo', (string) $uri);
     }
 
-    /**
-     * @expectedException \InvalidArgumentException
-     * @expectedExceptionMessage The path of a URI with an authority must start with a slash "/" or be empty
-     */
-    public function testRelativePathAndAuhorityIsInvalid()
+    public function testRelativePathAndAuhorityIsAutomagicallyFixed()
```

Comment:
Maybe leave the old test here as well (just like the commented exception). You can skip the test itself, or use `expect`method. Not sure if it's right, just an idea.